app/services/model/RejectionReason.scala

```diff
@@ -17,70 +17,43 @@
 package services.model
 
 import com.github.tototoshi.csv._
-import models.{ExportsDeclaration, Pointer}
+import config.AppConfig
+import javax.inject.{Inject, Singleton}
 import models.declaration.notifications.Notification
+import models.{ExportsDeclaration, Pointer}
 import play.api.Logger
 import play.api.i18n.Messages
 import play.api.libs.json.{Json, OFormat}
 
 import scala.io.Source
 
-case class RejectionReason(code: String, cdsDescription: String, exportsDescription: String, url: Option[String], pointer: Option[Pointer])
+case class RejectionReason(code: String, summaryErrorMessage: String, url: Option[String], pageErrorMessage: Option[String], pointer: Option[Pointer])
 
 object RejectionReason {
+  implicit val format: OFormat[RejectionReason] = Json.format[RejectionReason]
+  private val logger = Logger(this.getClass)
 
-  val allRejectionReasons: List[RejectionReason] = {
-    val reader =
-      CSVReader.open(Source.fromURL(getClass.getClassLoader.getResource("code-lists/errors-dms-rej-list.csv"), "UTF-8"))
-
-    val errors: List[List[String]] = reader.all()
-
-    errors.map(RejectionReason.apply)
+  def extractErrorDescription(useImprovedErrorMessage: Boolean, cdsDescription: String, exportsDescription: String): String = {
+    val useDescription = if (useImprovedErrorMessage) exportsDescription else cdsDescription
+    if (useDescription.isEmpty) cdsDescription else useDescription
```

Comment:
Not sure if I understand the logic correctly.
Is it the same below? Proposal:
```
val useDescription = if(useImprovedErrorMessage && exportsDescription.nonEmpty) exportsDescription else cdsDescription
```

Putting this in one if can be more readable, because it will show that use `exportsDescription` only when exists and feature flag is on

Reply:
Yes.  That is probably clearer.  I was just keeping a "flavour" of the existing methods.